st/clitests/awss3api_spec.py

```diff
@@ -1841,6 +1853,335 @@ def get_md5_sum(fname):
 #     .execute_test()\
 #     .command_is_successful()
 
+#*********************List Object Versions***************************
+#******** Create Bucket ********
+AwsTest('Aws can create bucket').create_bucket("versionlistbucket").execute_test().command_is_successful()
+
+AwsTest('Aws can enable versioning on bucket').put_bucket_versioning("versionlistbucket", "Enabled")\
+  .execute_test().command_is_successful()
+
+'''
+Create following keys (both, regular and heirarchical) into a bucket:
+----
+asdf
+bo0
+boo/0...boo/1
+boo#
+boo+
+fo0
+foo/0...foo/2
+foo#123
+foo+123
+qua0
+quax/0...quax/3
+quax#
+quax+
+-----
+'''
+# Step 0: Create directory with above structure.
+# Create temporary 's3listobjvertest' directory
+obj_list = []
+temp_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "s3listobjvertest")
+os.makedirs(temp_dir, exist_ok=True)
+# Clear contents of 's3listobjvertest' directory
+upload_content = temp_dir + "/*"
+os.system("rm -rf " + upload_content)
```

Comment:
Could simplify this with [tempfile.TemporaryDirectory](https://docs.python.org/3.6/library/tempfile.html#tempfile.TemporaryDirectory).

[Pathlib](https://docs.python.org/3.6/library/pathlib.html) is also nice and would simply things even more.

Reply:
We need a named directories and files here so that these could be used in the further stages. I could not find way to create named directories and use them/their path in the further stages. Keeping it as it is.
Please let me know if there is a way to achieve this.